chain/consensus/mir/manager.go

```diff
@@ -92,11 +93,16 @@ func NewManager(ctx context.Context,
 	}
 	id := cfg.Addr.String()
 
+	ts, err := node.ChainHead(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	if cfg.Consensus.SegmentLength < 0 {
 		return nil, fmt.Errorf("validator %v segment length must not be negative", id)
 	}
 
-	initialValidatorSet, err := membership.GetValidatorSet()
+	initialValidatorSet, err := membership.GetValidatorSet(string(netName), ts.String())
```

Comment:
We always query for the same network name, maybe we can set it in initialization so we don't have to propagate the arguments to all membership implementations.